src/core/analysis-context.ts

```diff
@@ -247,16 +247,26 @@ export class AnalysisContext {
   }
 
   /**
-   * Return a copy, but with the given cache.
+   * Returns a copy of the context but with optional replacements of cache or
+   * constructor options.
+   *
+   * Note: this feature is experimental.
    */
-  private _fork(cache: AnalysisCache): AnalysisContext {
-    const copy = new AnalysisContext({
+  _fork(cache?: AnalysisCache, options?: ForkOptions): AnalysisContext {
+    const contextOptions: Options = {
       lazyEdges: this._lazyEdges,
       parsers: this._parsers,
       scanners: this._scanners,
       urlLoader: this._loader,
-      urlResolver: this._resolver
-    });
+      urlResolver: this._resolver,
+    };
+    if (options) {
+      Object.assign(contextOptions, options);
+    }
+    const copy = new AnalysisContext(contextOptions);
+    if (!cache) {
+      cache = this._cache.invalidate([]);
+    }
     copy._cache = cache;
```

Comment:
If `options` has any keys besides urlLoader, I think we shouldn't copy over the cache. e.g. if the user adds a new scanner, then we'll need to at least re-scan, resolve, etc all documents. Copying over the right parts of the cache is an optimization we can get to later as needed.

I suppose you could make this argument for urlLoader too. Hm. Actually, for the bundler case, you do want to reload files from the incoming stream, don't you. You don't want to load from the cache from the previous urlLoader, do you?

I suppose you'll need to check which files have changed and call filesChanged() (will be made public in https://github.com/Polymer/polymer-analyzer/pull/545).

If this makes sense and is still a useful use case, it'll need some docs to let the user know of these constraints I think. I guess _fork is an experimental method, but we should at least have a marker in the docs indicating that the semantics of forking with options are hairy and we'll need to carefully vet them before exposing the _fork method.